pkg/asset/installconfig/azure/azure.go

```diff
@@ -105,8 +108,8 @@ func getRegions() (map[string]string, error) {
 	return allLocations, nil
 }
 
-func getResourceCapableRegions() ([]string, error) {
-	client, err := NewClient(context.TODO())
+func getResourceCapableRegions(cloudName azure.CloudEnvironment) ([]string, error) {
```

Comment:
It might make more sense for these (`getRegions` and `getResourceCapableRegions`) to live in the azure client API. In general I think anything that uses a client should follow this pattern, but I'm not very familiar with the Azure code.

Moving it to the API interface would allow mocking and testing these functions.  You didn't create these functions so just a suggestion.

Reply:
I do not agree that the functions make more sense as part of the client API. The functions rely on making calls to the client API and formatting the results in ways specific to what is needed by the Azure Platform. I do, however, take your point that it would be good to be able to mock and test the functions. So, in that vein, I will change the functions to take a client as a parameter rather than creating a client inside the function.